.github/workflows/delete_backport_branch.yml

```diff
@@ -7,9 +7,16 @@ on:
 jobs:
   delete-branch:
     runs-on: ubuntu-latest
+    permissions:
+      contents: write
     if: startsWith(github.event.pull_request.head.ref,'backport/')
     steps:
-      - name: Delete merged branch
-        uses: SvanBoxel/delete-merged-branch@main
-        env:
-          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+    - name: Delete merged branch
+      uses: actions/github-script@v7
+      with:
+        script: |
+          github.rest.git.deleteRef({
+            owner: context.repo.owner,
+            repo: context.repo.repo,
+            ref: `heads/${context.payload.pull_request.head.ref}`,
```

Comment:
which permission/role does the actions used to do deletion? previously, we have `GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}`

Reply:
It uses permissions as mentioned [here](https://github.com/opensearch-project/OpenSearch-Dashboards/pull/8351/files#diff-0153e43e8683726e844f93b477ceed014452fc9006230d868bb6210bbff54d2aR10-R11)
By default it uses {{GITHUB_TOKEN}} but we can use another PAT too if elevated permissions are required:
https://github.com/actions/github-script?tab=readme-ov-file#using-a-separate-github-token